helm/thingsboard/templates/node.yaml

```
@@ -65,6 +65,10 @@ spec:
           image: {{ include "thingsboard.postgresImage" . }}
           command: ['sh', '-c',
             'export PGPASSWORD=''{{ index .Values "postgresql-ha" "postgresql" "password" }}'' && export row_count=0 && until [ $row_count -ge 3 ]; do sleep 1; row_count=$(psql -h {{ include "thingsboard.pgpoolservicename" . }} -p {{ index .Values "postgresql-ha" "pgpool" "containerPorts" "postgresql" }} -U {{ index .Values "postgresql-ha" "postgresql" "username" }} -d {{ index .Values "postgresql-ha" "postgresql" "database" }} --set=on_error_stop=1 -t -c "SELECT count(*) FROM queue;"); done;']
+        - name: set-partitions-db-queue
+          image: {{ include "thingsboard.postgresImage" . }}
+          command: ['sh', '-c',
+            'export PGPASSWORD=''{{ index .Values "postgresql-ha" "postgresql" "password" }}'' && psql -h ''{{ include "thingsboard.pgpoolservicename" . }}'' -p {{ index .Values "postgresql-ha" "pgpool" "containerPorts" "postgresql" }} -U {{ index .Values "postgresql-ha" "postgresql" "username" }} -d {{ index .Values "postgresql-ha" "postgresql" "database" }} --set=on_error_stop=1 -t -c "UPDATE queue set partitions=2;"']
```

```suggestion
            'export PGPASSWORD=''{{ index .Values "postgresql-ha" "postgresql" "password" }}'' && psql -h ''{{ include "thingsboard.pgpoolservicename" . }}'' -p {{ index .Values "postgresql-ha" "pgpool" "containerPorts" "postgresql" }} -U {{ index .Values "postgresql-ha" "postgresql" "username" }} -d {{ index .Values "postgresql-ha" "postgresql" "database" }} --set=on_error_stop=1 -t -c "UPDATE queue set partitions={{ .Values.kafka.queues.numPartitions }};"']
```

How do we guarantee tb-node starts after this update?

Notice that we should also guarantee that mqtt server starts after the number of kafka queue partitions is updated.

We should also keep in sync the number of partitions of TB_CORE topic, it is still configured via env var, similar to EVP1.

---

As it is put as an init container I understand it guarantees that is done before the tb-node starts

---

right! thanks @i-castanedamalqui